app/repositories/user_repository.py

```diff
@@ -1,31 +1,48 @@
 import app.mappers.user_mapper as mapper
 from app import db
 from app.dtos.user_dto import UserDTO
+from app.exceptions import UserNotExistError
 from app.models.user import User
 
 
-def create_new_user(user_dto: UserDTO):
-    new_user = mapper.of_dto(user_dto)
-    db.session.add(new_user)
-    db.session.commit()
-
-
-def get_user_by_id(user_id: int) -> UserDTO:
-    user = User.query.filter_by(id=user_id).first()
-    return mapper.to_dto(user)
-
-
-def get_user_by_email(email: str) -> UserDTO:
-    user = User.query.filter_by(email=email).first()
-    return mapper.to_dto(user)
-
-
-def delete_user_by_id(user_id: int) -> None:
-    User.query.filter_by(id=user_id).delete()
-    db.session.commit()
-
-
-def update_user_password(user_id: int, new_password: str) -> None:
-    user = User.query.filter_by(id=user_id).first()
-    user.password = new_password
-    db.session.commit()
+class UserRepository:
+    @staticmethod
+    def create_new_user(user_dto: UserDTO):
+        new_user = mapper.of_dto(user_dto)
+        db.session.add(new_user)
+        db.session.commit()
+
+    @staticmethod
+    def get_user_by_id(user_id: int) -> UserDTO:
+        user = User.query.filter_by(id=user_id).first()
+        if user is None:
+            raise UserNotExistError
+        return mapper.to_dto(user)
+
+    @staticmethod
+    def get_user_by_email(email: str) -> UserDTO:
+        user = User.query.filter_by(email=email).first()
+        if user is None:
+            raise UserNotExistError
+        return mapper.to_dto(user)
+
+    @staticmethod
+    def delete_user_by_id(user_id: int) -> None:
+        user = User.query.filter_by(id=user_id).first()
+        if user is None:
+            raise UserNotExistError
+        User.query.filter_by(id=user_id).delete()
+        db.session.commit()
```

Comment:
Nie wiem czy rzucanie tutaj wyjątkiem to dobry pomysł. Ja to bym zwracał True jeżeli był w bazie i False jeżeli nie. I w sumie w takim ```get_user_by_email``` to bym zwracał None i obsługiwał to w innym miejscu. Ale to takie luźne przemyślenia, bo kiedyś coś słyszałem, że nie powinno się nadmiernie rzucać wyjątków

Reply:
Metody które zwracają false lub true powinny coś sprawdzać, a nie wykonywać operacje na bazie danych. Sytuacja, w której próbujemy usunać/pobrać użytkownika, który nie istnieje moim zdaniem jest sytuacją patologiczną i zasługuje na wyjątek. Zwróc uwagę, że faktycznie linijka User.query.filter_by(id=user_id).first() zwróci None jeśli nie ma usera o takim id i tu problemu nie ma, ale w         return mapper.to_dto(user) będize błąd bo wykonujemy operacje na None.

Reply:
Sprawdza czy usunęło użytkownika, czy nie. A wyjątki to raczej są do sytuacji nieprzewidywalnych i nieoczekiwanych. A sytuacja, że ktoś podał zły email to jednak jest bardzo przewidywalna. Ale no generalnie może to kwestia konwencji. Jak cię nie przekonałem to możesz kliknąć resolve, bo za dużo na wydajności nie straci program od tego.

Reply:
Jeśli chodzi o metody, które wyszukują użytkownika po id, przy błędnym id powinnien być zwracany wyjątek i obsłużony zostać w kontrolerze/serwisie i tak jest. Co do wyszukiwania po mailu, to mogę ewentualnie przenieść to do auth_service, jeśli będize user =None i rzucić inny wyjątek, który przechwyce w kontrolerze, ale moim zdaniem zwracanie None, a rzucenie wyjątku sprowadza się do tego samego i tak i tak to trzeba obsłużyć, a zwrócenie None może spowodować dziwne rzeczy przy wykoynwaniu operacji dla tego użytkownika. Przy wyjątkach do tego nie dojdzie, bo program przestanie działać, chyba że wyjąek zostanie przechwycony. Wyjątek w tym miejscu zapewnia że sytuacja gdzie nie znajdzeimy user'a będzie odpowiednio obsłużona, w przypadku zwrócenia None, jeśli zapomnimy tego obsłużyć to nie mamy pewności że zostanie to zasygnalizowane.